build-logic/buildquality/src/main/kotlin/gradlebuild.ci-reporting.gradle.kts

```diff
@@ -118,10 +113,26 @@ fun Task.findTraceJson(): List<File> {
     }
 }
 
-fun tmpTestFiles() =
-    layout.buildDirectory.dir("tmp/test files").get().asFile.listFiles()?.filter { dir ->
-        Files.walk(dir.toPath()).use { paths -> !paths.allMatch { it.toFile().isDirectory } }
-    } ?: emptyList()
+fun tmpTestFiles(): List<File> {
```

Comment:
I had something slightly different in mind, here: So you'd create a map from tmpDir to leftover files and use that going forward instead of this list of files.

Something like:
```kotlin
fun tmpTestFiles() =
    layout.buildDirectory.dir("tmp/test files").get().asFile.listFiles()?.associateWith { dir ->
        val dirPath = dir.toPath()
        Files.walk(dirPath).use { paths ->
            paths
                .filter { !it.toFile().isDirectory }
                .limit(4)
                .map { it.relativize(dirPath).toString() }
                .collect(Collectors.toList())
        }
    } ?: emptyMap()
```

And then do the reporting by using this map, i.e. later on you can do something like this to get a better error message:

```kotlin
        val nonEmptyDirs = tmpTestFiles.entries.joinToString("\n") { (dir, relativePaths) ->
            "${dir.absolutePath}:\n    ${relativePaths.joinToString("\n    ")}"
        }
```

Reply:
I applied your suggestion and also fixed issues in your code, then did a local test. PTAL.

The local test data:

```
            project.buildDir.resolve("tmp/test files").apply {
                resolve("a/b/c").mkdirs()
                resolve("a/b/c/d").createNewFile()
                resolve("a/b/e").mkdirs()
                resolve("1").mkdirs()
                resolve("1/2").createNewFile()
                resolve("3").mkdirs()
                resolve("4").createNewFile()
            }
```

Output:

```
Found non-empty test files dir:
/Users/zhb/Projects/gradle/subprojects/core/build/tmp/test files/1:
 2
/Users/zhb/Projects/gradle/subprojects/core/build/tmp/test files/a:
 b/c/d
/Users/zhb/Projects/gradle/subprojects/core/build/tmp/test files/4:
```